torchvision/datasets/flowers102.py

```diff
@@ -0,0 +1,119 @@
+        target_transform (callable, optional): A function/transform that takes in the target and transforms it.
+    """
+
+    download_url_prefix = "https://www.robots.ox.ac.uk/~vgg/data/flowers/102/"
+    file_dict = {
+        "image": ("102flowers.tgz", "52808999861908f626f3c1f4e79d11fa"),
+        "label": ("imagelabels.mat", "e0620be6f572b9609742df49c70aed4d"),
+        "setid": ("setid.mat", "a5357ecc9cb78c4bef273ce3793fc85c"),
+    }
+    splits_map = {"train": "trnid", "valid": "valid", "test": "tstid"}
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        download: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+    ) -> None:
+        super().__init__(root, transform=transform, target_transform=target_transform)
+        self._split = verify_str_arg(split, "split", ("train", "valid", "test"))
+        self._base_folder = Path(self.root) / "flowers-102"
+        self._images_folder = self._base_folder / "jpg"
+
+        if download:
+            self.download()
+
+        if not self._check_integrity():
+            raise RuntimeError("Dataset not found or corrupted. You can use download=True to download it")
+
+        from scipy.io import loadmat
+
+        # Read the label ids
+        label_mat = loadmat(self._base_folder / self.file_dict["label"][0])
+        labels = label_mat["labels"][0]
+
+        self.labels = np.unique(labels).tolist()
+        self.label_to_idx = dict(zip(self.labels, range(len(self.labels))))
+
+        # Read the image ids
+        set_ids = loadmat(self._base_folder / self.file_dict["setid"][0])
+        image_ids = set_ids[self.splits_map[self._split]][0]
+
+        self._labels = []
+        self._image_files = []
+
+        for image_id in image_ids:
+            self._labels.append(self.label_to_idx[labels[image_id - 1]])
+            self._image_files.append(self._images_folder / f"image_{image_id:05d}.jpg")
+
+    def __len__(self) -> int:
+        return len(self._image_files)
+
+    def __getitem__(self, idx) -> Tuple[Any, Any]:
+        image_file, label = self._image_files[idx], self._labels[idx]
+        image = PIL.Image.open(image_file).convert("RGB")
+
+        if self.transform:
+            image = self.transform(image)
+
+        if self.target_transform:
+            label = self.target_transform(label)
+
+        return image, label
+
+    def extra_repr(self) -> str:
+        return f"split={self._split}"
+
+    def _check_integrity(self):
+        st1 = check_integrity(str(self._base_folder / self.file_dict["image"][0]), self.file_dict["image"][1])
+        st2 = check_integrity(str(self._base_folder / self.file_dict["label"][0]), self.file_dict["label"][1])
+        st3 = check_integrity(str(self._base_folder / self.file_dict["setid"][0]), self.file_dict["setid"][1])
+        if not (st1 and st2 and st3):
+            return False
+        return True
```

Comment:
1. We should not check the image archive, since the user can delete it after we have extracted it and the dataset should still work.
2. Using a loop makes this more concise without reducing readability:

```suggestion
        for id in ["label", "setid"]:
            filename, md5 = self.file_dict[id]
            if not check_integrity(str(self._base_folder / filename), md5):
                return False
        return True
```